### PR TITLE
Add switches to build Control Gallery using the nuget produced by the build

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

This adds some build properties to our controls and control gallery projects so you can specify a nuget version you want to build the gallery against

The following command will make the android gallery use the xamarin forms version here instead of source code

```cmd
msbuild Xamarin.Forms.sln /t:restore  /p:XamarinFormsVersion=4.5.0.75
```

This also adds a nuget.config file pointing to our nightly feed and a directory to copy nugets into for testing against the CG

### Platforms Affected ### 
- Android

### Testing Procedure ###
- make sure the UI Tests run

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by mattleibow** [comment]
Hmm, going to need some other sprinkles too... My package is magic only for me, pushing an update: https://github.com/xamarin/XamarinComponents/pull/775

File: DevopsNuget.config [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <!-- add an Azure Artifacts feed -->
+    <!-- also get packages from the NuGet Gallery -->
```
Comment (mattleibow) [inline DevopsNuget.config]: If you really want to reset this, you might need to add `<clear />` at the top.

File: Stubs/Xamarin.Forms.Platform.iOS/Xamarin.Forms.Platform.iOS (Forwarders).csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">iPhoneSimulator</Platform>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
```
Comment (PureWeen) [inline Stubs/Xamarin.Forms.Platform.iOS/Xamarin.Forms.Platform.iOS (Forwarders).csproj]:
```suggestion
    <Platform Condition=" '$(Platform)' == '' ">iPhoneSimulator</Platform>
```

File: build/steps/build-osx.yml [hunk 3]
```diff
@@ -74,6 +92,7 @@ steps:
       Contents: |
        **/$(IpaName)
        Xamarin.Forms.Core.iOS.UITests/bin/$(BuildConfiguration)/**
+       nuget/**
```
Comment (PureWeen) [inline build/steps/build-osx.yml]:
```suggestion
```

File: build/steps/build-osx.yml [hunk 4]
```diff
@@ -51,12 +69,26 @@ steps:
...
+  - task: MSBuild@1
+    displayName: 'Build iOS Tests'
     inputs:
-      solutionFile: $(slnPath)
+      solution: 'Xamarin.Forms.Core.iOS.UITests/Xamarin.Forms.Core.iOS.UITests.csproj'
       configuration: $(buildConfiguration)
+      msbuildArguments: /bl:$(Build.ArtifactStagingDirectory)/ios-uitests.binlog
 
+  - task: MSBuild@1
+    displayName: 'Build iOS Tests'
```
Comment (PureWeen) [inline build/steps/build-osx.yml]:
```suggestion
    displayName: 'Build Android Tests'
```

**Comment by rmarinho** [comment]
Failing on ANdroid tests, seems doesn't start the app. iOS seems good.

**Comment by PureWeen** [comment]
@rmarinho yea it's not even starting on Android :-/ I emailed app center to give me some logs as to why so we'll see

**Pull request merged**